### DISCUSSION TITLE
uiButton 'keep button ON'

### DISCUSSION THREAD

**Discussion by DRBAUS** [RomanKubiak/ctrlr#283] [Q&A]
I noticed this 'keeping Button ON wil produce a continues stream of messages' in the Component part of the uiButton.

Has anyone ever used this and are those message definable? If so it could be very useful in combination with the 'Rate at which messages will be send, in milliseconds'.

**Comment by Ted-juh** [comment]
When you have a button that has a function to minimize or maximize the size of your panel, it could be handy to fire the button every few milliseconds. That way you don't have to click it 20 times to get the desired size. You could use the decremental/ incremental slider to do the same job but that also means you always have the -/+ buttons next to each other GUI-wise. What if you want them to be on top of each other? Then you can mimic the same effect with the continuous-on button.

**Reply by DRBAUS** [reply]
Ahaa. 

So it would be useless to think we could use it to send preprogrammed sysex messages in a timed way, I guess, like a sequence of notes for example.

**Reply by Ted-juh** [reply]
Tricky, if you ask me. In theory, yes. One way is to count the number of clicks. I don't know if that's possible. But if it is, then you could write a function that every time the number of clicks goes up it sends a certain note-on message. Or use getMiliSecondsSinceButtonDown to just send a different note-on message on a chosen interval.  Or if that doesn't work, run a timer on the luaModulatorValueChange method.

**Reply by DRBAUS** [reply]
Thanks for explaining!
I think I will need to get my head around using a Timer if I want to do this.

**Comment by Godlike-Productions** [comment]
Also useful for sending + or - button presses. Kurzweil's have a + and - button for incrementing and decrementing the currently selected value, for example, where I'll implement this.
